barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover_internal.hpp

```diff
@@ -391,42 +405,43 @@ template <class DeciderProvingKeys_> class ProtogalaxyProverInternal {
         // doesn't skip computation), so we need to define types depending on the template instantiation
         using ThreadAccumulators = TupleOfTuples;
         using ExtendedUnivatiatesType =
-            std::conditional_t<skip_zero_computations, ExtendedUnivariates, ExtendedUnivariatesNoOptimisticSkipping>;
+
+            std::conditional_t<Flavor::USE_SHORT_MONOMIALS,
```

Comment:
same question here as to whether we still do optimistic skipping

Reply:
I think that we do still do optimistic skipping. I tried to trace where this type is used, and it seems to only be used when initialising the `AllEntities` data structure passed into the `accumulate` method for our relations.

In our relation classes, the `Accumulator` type is derived from the types in `ContainerOverSubrelations`. This in turn is the same as the type `TupleOfTuplesOfUnivariates_` param in `accumulate_relation_univariates` from `protogalaxy_prover_impl.hpp`, which is the same as the `TupleOfTuples` param in `compute_combiner` which is, from what I can tell, the `TupleOfTuplesOfUnivariates` type defined in `protogalaxy_prover.hpp` and this type *does* support skipping

Phew. I might try to rationalise this a bit as I am not sure the template parameter propagation in the protogalaxy methods is required as the type used is defined in the class

Reply:
I just pushed a change which removes the template parameter types in `compute_combiner` and `accumulate_relation_univariates`. The only time we need to not optimitically skip seems to be in `combiner.test.cpp` (which is why we needed parametrisable types), so I created bespoke methods for `combiner.test.cpp` by defining a new class `PGInternalTest` that inherits from `ProtogalaxyProverInternal` and adds back in methods that enable not skipping.

While this is more lines of code overall, I think it makes `ProtogalaxyProverInternal` cleaner and easier to read, as there are fewer template parameter propagation chains to follow when examining the code. wdyt?